Allow to change account info via provisioning api

nickvergessen opened the pull request:
Fix #4158

nickvergessen committed:
Allow to change account info via provisioning api

Signed-off-by: Joas Schilling <coding@schilljs.com>

ArtificialOwl reviewed (approved):
:+1:

schiessle commented:
It seems like we have a lot of tests for this operation. Would it make sense to extend them to also test the newly added stuff?

nickvergessen commented:
> It seems like we have a lot of tests for this operation. Would it make sense to extend them to also test the newly added stuff?

Yes sure, I just wanted to have an "this is the way to go" confirmation from you before. But i guess your comment means you are fine with it?

rullzer commented:
Looks good. But yeah lets extend the tests.

nickvergessen commented on apps/provisioning_api/lib/Controller/UsersController.php:
```diff
@@ -291,11 +291,11 @@ protected function getUserData($userId) {
+		$data[AccountManager::PROPERTY_DISPLAYNAME] = $targetUserObject->getDisplayName();
+		$data[AccountManager::PROPERTY_PHONE] = $userAccount[AccountManager::PROPERTY_PHONE]['value'];
+		$data[AccountManager::PROPERTY_ADDRESS] = $userAccount[AccountManager::PROPERTY_ADDRESS]['value'];
+		$data[AccountManager::PROPERTY_WEBSITE] = $userAccount[AccountManager::PROPERTY_WEBSITE]['value'];
```
@nextcloud/android @nextcloud/ios In case you make use of `GET ocs/v2.php/cloud/users/<username>` the name of `webpage` was changed to `website`

nickvergessen commented:
Added functional tests and fixed the old ones, since they just assumed editing values would work. But in fact they failed because they tried to set multiple values at once and that did not work.

schiessle commented on apps/provisioning_api/lib/Controller/UsersController.php:
```diff
@@ -327,8 +327,13 @@ public function editUser($userId, $key, $value) {
 			// Editing self (display, email)
 			$permittedFields[] = 'display';
+			$permittedFields[] = AccountManager::PROPERTY_DISPLAYNAME;
 			$permittedFields[] = 'email';
```
you could also use `AccountManager::PROPERTY_DISPLAYNAME;`, just to be more consistent

nickvergessen replied:
PROPERTY_EMAIL, but yeah done

schiessle commented on apps/provisioning_api/lib/Controller/UsersController.php:
```diff
@@ -340,9 +345,14 @@ public function editUser($userId, $key, $value) {
+				$permittedFields[] = AccountManager::PROPERTY_DISPLAYNAME;
 				$permittedFields[] = 'quota';
 				$permittedFields[] = 'password';
 				$permittedFields[] = 'email';
```
same as above

MorrisJobke reviewed (approved):
Tested and works 👍
